File: decompiled/sources/com/hjq/widget/view/DrawableTextView.java
package com.hjq.widget.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import androidx.appcompat.widget.AppCompatTextView;
import com.hjq.widget.R;

public final class DrawableTextView extends AppCompatTextView {
    private int mDrawableHeight;
    private int mDrawableWidth;

    public DrawableTextView(Context context) {
        this(context, (AttributeSet) null);
    }

    public DrawableTextView(Context context, AttributeSet attributeSet) {
        this(context, attributeSet, 0);
    }

    public DrawableTextView(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
        TypedArray obtainStyledAttributes = context.obtainStyledAttributes(attributeSet, R.styleable.DrawableTextView);
        this.mDrawableWidth = obtainStyledAttributes.getDimensionPixelSize(R.styleable.DrawableTextView_drawableWidth, 0);
        this.mDrawableHeight = obtainStyledAttributes.getDimensionPixelSize(R.styleable.DrawableTextView_drawableHeight, 0);
        obtainStyledAttributes.recycle();
        refreshDrawablesSize();
    }

    public void setDrawableSize(int i, int i2) {
        this.mDrawableWidth = i;
        this.mDrawableHeight = i2;
        if (isAttachedToWindow()) {
            refreshDrawablesSize();
        }
    }

    public void setDrawableWidth(int i) {
        this.mDrawableWidth = i;
        if (isAttachedToWindow()) {
            refreshDrawablesSize();
        }
    }

    public void setDrawableHeight(int i) {
        this.mDrawableHeight = i;
        if (isAttachedToWindow()) {
            refreshDrawablesSize();
        }
    }

    public void setCompoundDrawables(Drawable drawable, Drawable drawable2, Drawable drawable3, Drawable drawable4) {
        super.setCompoundDrawables(drawable, drawable2, drawable3, drawable4);
        if (isAttachedToWindow()) {
            refreshDrawablesSize();
        }
    }

    public void setCompoundDrawablesRelative(Drawable drawable, Drawable drawable2, Drawable drawable3, Drawable drawable4) {
        super.setCompoundDrawablesRelative(drawable, drawable2, drawable3, drawable4);
        if (isAttachedToWindow()) {
            refreshDrawablesSize();
        }
    }

    private void refreshDrawablesSize() {
        if (this.mDrawableWidth != 0 && this.mDrawableHeight != 0) {
            Drawable[] compoundDrawables = getCompoundDrawables();
            if (compoundDrawables[0] == null && compoundDrawables[1] == null) {
                Drawable[] compoundDrawablesRelative = getCompoundDrawablesRelative();
                super.setCompoundDrawablesRelative(limitDrawableSize(compoundDrawablesRelative[0]), limitDrawableSize(compoundDrawablesRelative[1]), limitDrawableSize(compoundDrawablesRelative[2]), limitDrawableSize(compoundDrawablesRelative[3]));
                return;
            }
            super.setCompoundDrawables(limitDrawableSize(compoundDrawables[0]), limitDrawableSize(compoundDrawables[1]), limitDrawableSize(compoundDrawables[2]), limitDrawableSize(compoundDrawables[3]));
        }
    }

    private Drawable limitDrawableSize(Drawable drawable) {
        int i;
        if (drawable == null) {
            return null;
        }
        int i2 = this.mDrawableWidth;
        if (!(i2 == 0 || (i = this.mDrawableHeight) == 0)) {
            drawable.setBounds(0, 0, i2, i);
        }
        return drawable;
    }
}

File: decompiled/sources/com/hjq/widget/view/DrawableTextView.kt
package com.hjq.widget.view

import android.content.Context
import android.content.res.TypedArray
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.hjq.widget.R

class DrawableTextView : AppCompatTextView {
    private var mDrawableHeight = 0
    private var mDrawableWidth = 0

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        val typedArray: TypedArray = context.obtainStyledAttributes(attrs, R.styleable.DrawableTextView)
        mDrawableWidth = typedArray.getDimensionPixelSize(R.styleable.DrawableTextView_drawableWidth, 0)
        mDrawableHeight = typedArray.getDimensionPixelSize(R.styleable.DrawableTextView_drawableHeight, 0)
        typedArray.recycle()
        refreshDrawablesSize()
    }

    fun setDrawableSize(drawableWidth: Int, drawableHeight: Int) {
        mDrawableWidth = drawableWidth
        mDrawableHeight = drawableHeight
        if (isAttachedToWindow) {
            refreshDrawablesSize()
        }
    }

    fun setDrawableWidth(drawableWidth: Int) {
        mDrawableWidth = drawableWidth
        if (isAttachedToWindow) {
            refreshDrawablesSize()
        }
    }

    fun setDrawableHeight(drawableHeight: Int) {
        mDrawableHeight = drawableHeight
        if (isAttachedToWindow) {
            refreshDrawablesSize()
        }
    }

    override fun setCompoundDrawables(
        left: Drawable?,
        top: Drawable?,
        right: Drawable?,
        bottom: Drawable?
    ) {
        super.setCompoundDrawables(left, top, right, bottom)
        if (isAttachedToWindow) {
            refreshDrawablesSize()
        }
    }

    override fun setCompoundDrawablesRelative(
        start: Drawable?,
        top: Drawable?,
        end: Drawable?,
        bottom: Drawable?
    ) {
        super.setCompoundDrawablesRelative(start, top, end, bottom)
        if (isAttachedToWindow) {
            refreshDrawablesSize()
        }
    }

    private fun refreshDrawablesSize() {
        if (mDrawableWidth != 0 && mDrawableHeight != 0) {
            val drawables = compoundDrawables
            if (drawables[0] == null && drawables[1] == null) {
                val relativeDrawables = compoundDrawablesRelative
                super.setCompoundDrawablesRelative(
                    limitDrawableSize(relativeDrawables[0]),
                    limitDrawableSize(relativeDrawables[1]),
                    limitDrawableSize(relativeDrawables[2]),
                    limitDrawableSize(relativeDrawables[3])
                )
                return
            }
            super.setCompoundDrawables(
                limitDrawableSize(drawables[0]),
                limitDrawableSize(drawables[1]),
                limitDrawableSize(drawables[2]),
                limitDrawableSize(drawables[3])
            )
        }
    }

    private fun limitDrawableSize(drawable: Drawable?): Drawable? {
        if (drawable == null) {
            return null
        }
        if (mDrawableWidth != 0 && mDrawableHeight != 0) {
            drawable.setBounds(0, 0, mDrawableWidth, mDrawableHeight)
        }
        return drawable
    }
}
